daily_data/show_data.html.erb

```diff
@@ -3,10 +3,10 @@
  <% end %>
  </ul>
 
- <form action="/add/data" method="post" enctype="application/x-www-form-urlencoded">
+ <%= form_tag("/add/data", method: :post) do -%>
    <p><label>Date <input type="date" name="date"></label></p>
    <p><label>Step Count <input type="number" name="step_count"></label></p>
    <p><label>Notes <textarea name="notes" rows="5"></textarea></label></p>
 
    <p><button>Submit daily data</button></p>
- </form>
+<% end -%>
```

Comment:
You don’t need the `-` before the `%>` unless you especially care about trimming trailing whitespace after the tag, which I don’t expect you do in this case.